tests/unit/backend/layers/business/test_business.py

```diff
@@ -448,7 +461,20 @@ def test_add_dataset_to_unpublished_collection_ok(self):
         new_dataset_version = self.database_provider.get_dataset_version(new_dataset_version_id)
         self.assertIsNotNone(new_dataset_version)
         self.assertIsNone(new_dataset_version.metadata)
-        self.assertEqual(new_dataset_version.processing_status, DatasetStatus.WAITING)
+        self.assertEqual(new_dataset_version.status.upload_status, DatasetUploadStatus.WAITING)
```

Comment:
I forgot that the dataset status actually has multiple states (validation, processing, etc), so I had to overhaul these tests.